src/tests/ftest/util/agent_utils.py

```diff
@@ -97,6 +97,45 @@ def get_params(self, test):
         # Run daos_agent with test variant specific log file names if specified
         self.yaml.update_log_file(getattr(test, "agent_log"))
 
+    def get_sub_command_class(self):
+        """Get the daos_agent sub command object based on the sub-command."""
+        if self.sub_command.value == "dump-attachinfo":
+            self.sub_command_class = self.DumpAttachInfoSubCommand()
+        else:
+            self.sub_command_class = None
+
+    class DumpAttachInfoSubCommand(CommandWithSubCommand):
+        """Defines an object for the daos_agent dump-attachinfo sub command."""
+
+        def __init__(self):
+            """Create a daos_agent dump-attachinfo subcommand object."""
+            super(DaosAgentCommand.DumpAttachInfoSubCommand, self).__init__(
+                "/run/daos_agent/dump-attachinfo/*", "dump-attachinfo")
+
+            self.output = FormattedParameter("--output {}", None)
+
+    def dump_attachinfo(self, path="uri.txt"):
+        """Write CaRT attachinfo file
+
+        Args:
+            path (str): Path to attachinfo file.
+
+        Returns:
+            CmdResult: Object that contains exit status, stdout, and other
+                information.
+
+        Raises:
+            CommandFailure: if the daos_agent dump-attachinfo command fails.
+
+        """
+        self.set_sub_command("dump-attachinfo")
+        self.sub_command_class.output.value = path
+        try:
+            return self.run()
+        except CommandFailure as error:
+            raise CommandFailure(
+                "<daos_agent> dump-attachinfo failed: {}".format(error))
```

Comment:
Since `DaosCommand` and `DmgCommand` both have basically the same definition for [`_get_result()`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/daos_utils.py#L481) - which is the same as this code here - it should be a common method in [`CommandWithSubCommand`](https://github.com/daos-stack/daos/blob/1812929e27424c457c816691abefd6b4e1048e7c/src/tests/ftest/util/command_utils.py#L287).  The one change that would be required is this line:
```
            raise CommandFailure("<{}> command failed: {}".format(self.command, error))
```

Then this change could be applied:
```suggestion
        return self._get_result()
```

Reply:
Hmm. I tried this, but `CommandWithSubcommand` doesn't have the yaml stuff, so I get 'undefined reference to self.yaml', etc.

Reply:
Trying something else...

Reply:
I put a `_get_result()` method on `YamlCommand` that does the yaml stuff and then uses `super()` to call the parent method. Seems to work fine in my local testing.

Reply:
Hmm ... `DaosAgentCommand` inherits `YamlCommand` which inherits `SubProcessCommand` which inherits `CommandWithSubCommand`, so placing the `def _get_result(self)` in `CommandWithSubCommand` should work.